app/models/game.rb

```diff
@@ -8,4 +8,30 @@ class Game < ApplicationRecord
           OR
           (white_player_id IS NOT NULL AND black_player_id IS NULL)')
         }
+
+  class Board
+    attr_reader :grid
+    
+    def initialize(pieces)
+      @pieces = pieces
+      @grid = Array.new(8) { Array.new(8) }
+      fill_grid
+    end
+    
+    private
+    
+    def pieces
```

Comment:
A cool Ruby shortcut: `attr_reader`, `attr_writer`, and `attr_accessor` all just define method(s) when you call them - they're essentially just saving typing space.  As such, you can create this private getter by using `attr_reader` and because it's under `private`, it's private just like a method:

```ruby
# This one's public
attr_reader :attribute

private

# This one's private
attr_reader :attribute
```